service/src/test/kotlin/no/nav/su/se/bakover/service/revurdering/IverksettRevurderingTest.kt

```diff
@@ -1,190 +1,325 @@
-                },
-            )
-            verify(revurderingRepoMock).defaultTransactionContext()
-            verify(revurderingRepoMock).lagre(argThat { it shouldBe iverksattRevurdering }, anyOrNull())
-            verify(eventObserver, times(2)).handle(any())
-        }
-        verifyNoMoreInteractions(
-            revurderingRepoMock,
-            utbetalingServiceMock,
-            utbetalingMock,
+        val simulertUtbetaling = simulertUtbetaling()
+
+        val serviceAndMocks = RevurderingServiceMocks(
+            revurderingRepo = mock {
+                on { hent(any()) } doReturn revurderingTilAttestering
+                doNothing().whenever(it).lagre(any(), anyOrNull())
+            },
+            utbetalingService = mock {
+                on { genererUtbetalingsRequest(any(), any(), any(), any(), any()) } doReturn simulertUtbetaling.right()
+                on { publiserUtbetaling(any()) } doReturn Utbetalingsrequest("<xml></xml>").right()
+            },
+            vedtakRepo = mock {
+                doNothing().whenever(it).lagre(any(), anyOrNull())
+            }
+        )
+
+        val respone = serviceAndMocks.revurderingService.iverksett(revurderingTilAttestering.id, attestant)
+
+        respone shouldBe expected.right()
+
+        verify(serviceAndMocks.revurderingRepo).hent(argThat { it shouldBe revurderingTilAttestering.id })
+        verify(serviceAndMocks.utbetalingService).genererUtbetalingsRequest(
+            sakId = argThat { it shouldBe revurderingTilAttestering.sakId },
+            attestant = argThat { it shouldBe attestant },
+            beregning = argThat { it shouldBe revurderingTilAttestering.beregning },
+            simulering = argThat { it shouldBe revurderingTilAttestering.simulering },
+            uføregrunnlag = argThat { it shouldBe revurderingTilAttestering.vilkårsvurderinger.uføre.grunnlag },
         )
+        verify(serviceAndMocks.utbetalingService).lagreUtbetaling(argThat { it shouldBe simulertUtbetaling }, anyOrNull())
+        verify(serviceAndMocks.utbetalingService).publiserUtbetaling(argThat { it shouldBe simulertUtbetaling })
+        verify(serviceAndMocks.vedtakRepo).lagre(
+            argThat { it should beOfType<VedtakSomKanRevurderes.EndringIYtelse.InnvilgetRevurdering>() },
+            anyOrNull()
+        )
+        verify(serviceAndMocks.revurderingRepo).lagre(argThat { it shouldBe expected }, anyOrNull())
+
+        serviceAndMocks.verifyNoMoreInteractions()
     }
 
     @Test
     fun `iverksett - iverksetter opphør av ytelse`() {
-        val (_, revurderingTilAttestering) = tilAttesteringRevurderingOpphørtUføreFraInnvilgetSøknadsbehandlingsVedtak()
-        val utbetaling = oversendtUtbetalingUtenKvittering()
-        val mocks = RevurderingServiceMocks(
+        val revurderingTilAttestering = tilAttesteringRevurderingOpphørtUføreFraInnvilgetSøknadsbehandlingsVedtak().second
+        val simulertUtbetaling = simulertUtbetaling()
```

Comment:
Det finnes egne varianter som gir simuleringe for stans/gjenopptak/opphør også som kan kan vurdere å bruke. Må vel vurdres opp mot at man ikke ender opp med å gjøre tåpelige asserts som `a == a` etc?